_ingest-pipelines/processors/dissect.md

```diff
@@ -0,0 +1,289 @@
+
+`%{key1} %{key2}`
+
+The output is:
+
+`value1 value2`
+
+You can also specify a custom separator using the `append_separator` parameter. For example, the following pattern uses a comma as the separator: 
+
+`%{key1} %{key2}, append_separator => ","`
+
+The output is:
+
+`value1, value2`
+
+### Append with order modifier (`+` and `/`)
+
+The append with order modifier combines the values of two or more keys into a single output value based on the order specified after the `/`. You have the flexibility to customize the separator that separates the appended values. The append modifier is useful for compiling multiple fields into a single formatted output line, constructing structured lists of data items, and consolidating values from various sources.  
+
+#### Example of usage
+
+The following example pipeline uses the append with order modifier. Note that the `append_separator` parameter must be defined in the processor configuration, outside of the `pattern`. It is only relevant with the `+` modifier. See the following example pipeline:
+
+```json
+PUT /_ingest/pipeline/dissect-test
+{
+  "description": "Pipeline that dissects web server logs",
+  "processors": [
+    {
+      "dissect": {
+        "field": "message",
+        "pattern": "%{a} %{+a} %{+a}",
+        "append_separator":  ","
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+If you ingest the following example document, you'll get the response `"a":"apple,banana,coconut"`:
+
+```json
+{
+  "message": "apple banana coconut"
+}
+```
+{% include copy-curl.html %}
+
+### Named skip key
+
+The named skip key modifier excludes specific matches from the final output by using an empty key `{}` or `?` modifier within the pattern. For example, the following patterns are equivalent: `%{firstName} %{lastName} %{?ignore}` and `%{firstName} %{lastName} %{}`. The named skip key modifier is useful for excluding irrelevant or unnecessary fields from the output, focusing on specific information, or streamlining the output for further processing or analysis. 
+
+#### Example of usage
+
+The following pattern uses a named skip key to exclude a field (in this case, `ignore`) from the output. You can assign a descriptive name to the empty key, for example, `%{?ignore}`, to clarify that the corresponding value should be excluded from the final result:
+
+`%{firstName} %{lastName} %{?ignore}`
+
+### Reference keys (`*` and `&`)
+
+Reference keys use parsed values as key/value pairings for structured content. This can use useful when handling systems that partially log data in key/value pairs. By using reference keys, you can preserve the key/value relationship and maintain the integrity of the extracted information. 
+
+#### Example of usage
+
+The following pattern uses a reference key to extract data into a structured format. In this example, `%{value}` represents the parsed value and `%{reference_key}` acts as the placeholder for the actual key:
```

Comment:
Here's an example:
extract `client_ip` and two key/value pairs for the next values:
```
PUT /_ingest/pipeline/dissect-test
{
  "description": "Pipeline that dissects web server logs",
  "processors": [
    {
      "dissect": {
        "field": "message",
        "pattern": "%{client_ip} %{*a}:%{&a} %{*b}:%{&b}"
      }
    }
  ]
}
```
Test the pipeline:
```
POST _ingest/pipeline/dissect-test/_simulate
{
  "docs": [
    {
      "_index": "testindex1",
      "_id": "1",
      "_source": {
        "message": "192.168.1.10 response_code:200 response_size:3456"
      }
    }
  ]
}
```
The two key/value pairs were extracted into fields:
```
{
  "docs": [
    {
      "doc": {
        "_index": "testindex1",
        "_id": "1",
        "_source": {
          "client_ip": "192.168.1.10",
          "response_code": "200",
          "message": "192.168.1.10 response_code:200 response_size:3456",
          "response_size": "3456"
        },
        "_ingest": {
          "timestamp": "2024-01-22T22:48:51.475535635Z"
        }
      }
    }
  ]
}
```

Reply:
Thank you for the examples.